ac.rkt

```diff
@@ -1035,10 +1036,6 @@
 
 (xdef swrite (lambda args (printwith write args)))
 (xdef disp  (lambda args (printwith display args)))
-
-; a special end-of-file uninterned symbol guaranteed never to be equal to the
-; result of any call to `(read)` or similar.
-(define eof (gensym 'eof))
```

Comment:
Deleting this breaks the `xdef` on the next line.

The reason `eof` exists: sometimes `(read)` can genuinely return `nil`, and we don't want it to be misinterpreted as end-of-file. So the idiom we've gradually settled on is:

```
(whiler e (read) eof
   ...)
```

And similarly for other control constructs besides `whiler`. You can see this sort of usage across the repo if you grep for `eof`.

Reply:
With the new changes, I'm not sure it would be:

```
arc> eof
#<eof>
arc> (no eof)
arc> 
```
The read functions seem to default their eof argument to the xdefined `eof` value.

Reply:
(Another way to phrase this is that racket's `eof` can be thought of as a unique symbol name, since `(type eof)` now returns `sym` anyway.)

Reply:
We could, but then it would be readable if it were accidentally written to disk.  The current behavior seems to guard against that:

```
arc> (tostring (prn eof))
"#<eof>\n"
arc> (fromstring that (read))
UNKNOWN::0: read: bad syntax `#<'
  context...:
   /Users/bb/Applications/Racket v6.12/collects/racket/private/kw.rkt:446:14: zz
```

Reply:
(I've updated the PR with some improvements. The main one is that Ctrl-C no longer exits, so I can switch `eof` back to a gensym if it might be better.)

Reply:
Looks like our comments crossed here. Let me spend some time digesting your comments above

Reply:
Oh wow, so Racket has [an existing symbol called `eof`](https://docs.racket-lang.org/reference/port-ops.html#%28def._%28%28quote._~23~25kernel%29._eof%29%29) that already does exactly the same thing? I had no idea. Ok, I'm convinced on this comment thread.